pinot-segment-local/src/main/java/org/apache/pinot/segment/local/upsert/BasePartitionUpsertMetadataManager.java

```diff
@@ -395,6 +402,18 @@ protected void finishOperation() {
     }
   }
 
+  /**
+   * When TTL is enabled for upsert, this function is used to remove expired keys from the primary key indexes.
```

Comment:
When will this method be invoked? It seems that it is not used anywhere except in tests.

Reply:
@chenboat this will be integrated once #10928 merged.